tests/foreman/cli/test_oscap.py

```diff
@@ -599,6 +605,36 @@ def test_positive_associate_scap_policy_with_hostgroups(self, scap_content, modu
             }
         )
         assert scap_policy['hostgroups'][0] == hostgroup['name']
+        name2 = gen_string('alphanumeric')
+        scap_policy2 = module_target_sat.cli_factory.make_scap_policy(
+            {
+                'name': name2,
+                'deploy-by': 'ansible',
+                'scap-content-id': scap_content["scap_id"],
+                'scap-content-profile-id': scap_content["scap_profile_id"],
+                'period': OSCAP_PERIOD['weekly'].lower(),
+                'weekday': OSCAP_WEEKDAY['friday'].lower(),
+                'hostgroups': hostgroup['name'],
+            }
+        )
+        assert scap_policy2['hostgroups'][0] == hostgroup['name']
+        module_target_sat.cli.HostGroup.delete({'id': hostgroup['id']})
+        # removal of hostgroup shouldn't affect policies
+        try:
+            module_target_sat.cli.Scappolicy.list()
+        except CLIReturnCodeError:
+            pytest.fail("failed to list policies")
```

Comment:
Why are we catching the failure?

Reply:
I believe that plain raising of the exception results in `Error` junit status. If you properly fail it, you get `Failed` status.

Reply:
True, `CLIReturnCodeError` doesn't inherit from `AssertionError` so the test would result as `Error`. And unfortunately pytest doesn't have anything like `pytest.not_raises()` (yet), which would be cool for these cases.